### PR TITLE
Adjust PAWN function to support 1D matrix for convenience

### PR DESCRIPTION
**Pull request by ConnectedSystems**
Say a user was assessing the mean of some scenario outcome

```julia
μ_tac = mean(ADRIA.metrics.scenario_total_cover(rs), dims=:timesteps)
ADRIA.sensitivity.pawn(rs, μ_tac)
```

Previously, this would fail with an error as `μ_tac` would be a matrix (i.e., $1 \times N$) rather than a vector which the method expects.

The change also supports transposed matrices:

```julia
μ_tac = mean(ADRIA.metrics.scenario_total_cover(rs), dims=:timesteps)'  # note the transpose!
ADRIA.sensitivity.pawn(rs, μ_tac)
```

The above `μ_tac` would be an $N \times 1$ matrix (again, instead of a vector).

The PR supports implicit conversion to vector simply for convenience.

Originally, the user was expected to convert to vector explicitly prior to use:

```julia
μ_tac = vec(mean(ADRIA.metrics.scenario_total_cover(rs), dims=:timesteps))
```

which is a little tedious to remember to do every time.

This PR also expands and updates the docstring to detail all methods and provide additional contextual information of the method.

### CODE REVIEW

File: src/analysis/sensitivity.jl [hunk 1]
```diff
@@ -53,8 +69,29 @@ NamedDimsArray, of min, mean, median, max, std, and cv summary statistics.
...
+
+Puy et al., found that the ratio of \$N\$ (number of samples) to \$S\$ has to be sufficiently high
+(\$N/S > 80\$) to avoid biased results. Pianosi and Wagener point  out this requirement is not
+particularly difficult to meet. Using the recommended value (\$S := 10\$), a sample of 1024 runs
+(small for purposes of Global Sensitivity Analysis) meets this requirement (\$1024/10 = 102.4\$).
+Additionally, lower values of \$N/S\$ is more an indication of faulty experimental design moreso
+than any deficiency of the PAWN method.
 """
-function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDimsArray where {T1<:AbstractMatrix{<:Real},T2<:AbstractVector{<:Real}}
+function pawn(X::AbstractMatrix{<:Real}, y::AbstractVector{<:Real}, factor_names::Vector{String}; S::Int64=10)::NamedDimsArray
```
Comment (Zapiano) [inline src/analysis/sensitivity.jl:94]: Following the style guide, this line shouldn't be longer than 92 characters

File: src/analysis/sensitivity.jl [hunk 2]
```diff
@@ -96,13 +133,23 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
...
 end
-function pawn(X::NamedDimsArray, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::NamedDimsArray, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
     return pawn(X, y, axiskeys(X, 2); S=S)
 end
-function pawn(rs::ResultSet, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::Union{DataFrame,AbstractMatrix{<:Real}}, y::AbstractMatrix{<:Real}; S::Int64=10)::NamedDimsArray
+    N, D = size(y)
+    if N > 1 && D > 1
+        throw(ArgumentError("The current implementation of PAWN can only assess a single quantity of interest at a time."))
```
Comment (Zapiano) [inline src/analysis/sensitivity.jl:145]: Again, line longer than 92 characters

File: src/analysis/sensitivity.jl [hunk 3]
```diff
@@ -96,13 +133,23 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
...
-function pawn(X::DataFrame, y::AbstractVector; S::Int64=10)::NamedDimsArray
+function pawn(X::DataFrame, y::AbstractVector{<:Real}; S::Int64=10)::NamedDimsArray
     return pawn(Matrix(X), y, names(X); S=S)
 end
-function pawn(X::NamedDimsArray, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::NamedDimsArray, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
     return pawn(X, y, axiskeys(X, 2); S=S)
 end
-function pawn(rs::ResultSet, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::Union{DataFrame,AbstractMatrix{<:Real}}, y::AbstractMatrix{<:Real}; S::Int64=10)::NamedDimsArray
```
Comment (Zapiano) [inline src/analysis/sensitivity.jl:142]: What happens if `X` and `y` are both `AbstractMatrix`? Then this function will call pawn passing `X::AbstractMatrix` and `y::Vector` without `factor_names`, right? Am I missing something or that won't work?

Comment (ConnectedSystems) [reply src/analysis/sensitivity.jl:142]: You're right - it should constrained to `NamedDimsArray`, not `AbstractMatrix`.

File: src/analysis/sensitivity.jl (hunk 1 above)
Comment (ConnectedSystems) [reply src/analysis/sensitivity.jl]: Yes okay - I had 100 as the max upper but happy to conform to 92.

**Comment by ConnectedSystems** [comment]
All done, ready for second review.

File: src/analysis/sensitivity.jl [hunk 4]
```diff
@@ -88,21 +132,43 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
...
-    return NamedDimsArray(results; factors=Symbol.(factor_names), Si=[:min, :mean, :median, :max, :std, :cv])
+    col_names = [:min, :mean, :median, :max, :std, :cv]
+    return NamedDimsArray(results; factors=Symbol.(factor_names), Si=col_names)
 end
-function pawn(X::DataFrame, y::AbstractVector; S::Int64=10)::NamedDimsArray
+function pawn(X::DataFrame, y::AbstractVector{<:Real}; S::Int64=10)::NamedDimsArray
     return pawn(Matrix(X), y, names(X); S=S)
 end
-function pawn(X::NamedDimsArray, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::NamedDimsArray, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
```
Comment (Zapiano) [inline src/analysis/sensitivity.jl:154]: Function name longer than 92 characters

File: src/analysis/sensitivity.jl [hunk 5]
```diff
@@ -88,21 +132,43 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
...
     return pawn(X, y, axiskeys(X, 2); S=S)
 end
-function pawn(rs::ResultSet, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::Union{DataFrame,NamedDimsArray}, y::AbstractMatrix{<:Real}; S::Int64=10)::NamedDimsArray
```
Comment (Zapiano) [inline src/analysis/sensitivity.jl:157]: Function name longer than 92 characters

File: src/analysis/sensitivity.jl [hunk 6]
```diff
@@ -88,21 +132,43 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
...
+            " of interest at a time."
+        )
+        throw(ArgumentError(msg))
+    end
+
+    # The wrapped call to `vec()` handles cases where matrix-like data type is passed in
+    # (N x 1 or 1 x D) and so ensures a vector is passed along
+    return pawn(X, vec(y); S=S)
+end
+function pawn(rs::ResultSet, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
```
Comment (Zapiano) [inline src/analysis/sensitivity.jl:171]: Function name longer than 92 characters

**Review by Zapiano — COMMENTED** [review]
There are three more function names longer than 92 characters. Other than that, LGTM

File: src/analysis/sensitivity.jl (hunk 6 above)
Comment (ConnectedSystems) [reply src/analysis/sensitivity.jl]: Obviously I'm getting old and going blind.

**Comment by ConnectedSystems** [comment]
I think I've fixed all line length issues, let me know if I missed anything again.

**Review by Zapiano — APPROVED** [review]

**Pull request merged**